src/core/transaction.c

```diff
@@ -396,23 +397,33 @@ static int transaction_verify_order_one(Transaction *tr, Job *j, Job *from, unsi
 
                 unit_ids = merge_unit_ids(unit_log_field(j->unit), array); /* ignore error */
 
+                _cleanup_free_ char *cycle_path_text = NULL;
                 STRV_FOREACH_PAIR(unit_id, job_type, array)
-                        /* logging for j not k here to provide a consistent narrative */
-                        log_struct(LOG_WARNING,
-                                   LOG_UNIT_MESSAGE(j->unit,
-                                                    "Found %s on %s/%s",
-                                                    unit_id == array ? "ordering cycle" : "dependency",
-                                                    *unit_id, *job_type),
+                        (void) strextendf_with_separator(
+                                        &cycle_path_text,
+                                        "; ",
```

Comment:
I find it a bit less readable when squashed into a single line. (What's our reason for avoiding multi-line log entries again?)

Reply:
I want this as one log-event, to retain context as one thing, and that we can attach one MESSAGE_ID to it, and fine each occasion just once. Now, we could fire that as multi-line log event, but log_struct() doesn't make that easy: if you pass a string with newlines to log_struct() it understands this as seting multiple fields, and that's unfortunately load-bearing (the unit_ids string in this very function relies on this), hence not easy to fix. Hence I opted to just turn this into a single-line log msg for now...

Reply:
Then rather than concating Found ...; messages together, rephrase it as "Found ordering cycle on ...; has dependencies a/start, b/start, ..."?